prediction-insights.ipynb

```diff
@@ -0,0 +1,774 @@
+       "6               1.123037  \n",
+       "7               1.326493  "
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "inf_case_indices = influential_cases[[\".session\", \".session_training_index\"]].values.tolist()\n",
+    "\n",
+    "anom_df = t.get_cases(\n",
+    "    case_indices=inf_case_indices,\n",
+    "    features=[\"sepal-length\", \"sepal-width\", \"petal-length\", \"petal-width\", \"target\", \"similarity_conviction\"]\n",
+    ")\n",
+    "\n",
+    "anom_df.sort_values(by=\"similarity_conviction\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Insight 3: Explanation\n",
+    "\n",
+    "Howso also provides detailed explanations about each prediction. Commonly used explanations are:\n",
+    "\n",
+    "- Feature Contributions for measuring feature importance\n",
+    "- Residuals for measuring the uncertainty of the local data\n",
+    "- Conviction for measurements that are relative to the expected value\n",
+    "\n",
+    "Feature contributions will help determine how much each context feature contributed to the prediction, on the scale\n",
+    "of the action feature."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'petal-width': 0.28871181572340304,\n",
+       " 'petal-length': 0.2931029454393371,\n",
+       " 'sepal-length': 0.04494568151237635,\n",
+       " 'sepal-width': 0.15656876988315238}"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "reaction[\"details\"][\"feature_contributions\"][0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Residuals can characterize the uncertainty of the data around the prediction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'petal-width': 0.25238915042846627,\n",
+       " 'petal-length': 0.6211629574817258,\n",
+       " 'target': 0.12239583333333334,\n",
+       " 'sepal-length': 0.5305952186638815,\n",
+       " 'sepal-width': 0.3249440776128855}"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "reaction[\"details\"][\"feature_residuals\"][0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Conviction refers to a number of statistics that are defined as:\n",
+    "\n",
+    "$$\n",
+    "\\text{Conviction} = \\frac{\\text{Measured Surprisal}}{\\text{Expected Surprisal}}\n",
+    "$$\n",
+    "\n",
+    "Since Howso provides numerous privacy measures, there are numerous types of conviction as well, such as similarity\n",
```

Comment:
No need to define all the types of conviction. LEt's just them to get results, then explain the results.